centering the "add new email address" form

colmoneill opened the pull request:
Tiny tiny change, as pointed out in #567; centring the new email form to the middle of the viewport.

Given the 'absolute positioning' nature of this app, I'm centring to the middle of the browser, not the middle of the left pannel. The difference being that on first launch of the Mail app in oC, you get this form prompt with no left menu, as the screenshot in #567 shows.

irgendwie commented:
If you already have an account and add an additional one (settings on the bottom left), then the form will not be centered.
In this case, this left menu is visible and therefore you can't center by the browser size.

colmoneill commented:
You are right, this was not compatible with a scenario that had an account set up already. It now is, but I had to change the structure of index.php for the mail-setup form to be included into the app-content div, for cascading styles to work properly.

The form now aligns to the centers in both cases. I did not thoroughly read through all the relative structure of the the index.php, so please review.

Cheers!

clementhk commented:
I like those works that were started as "tiny tiny change", then started to roll bigger and bigger.

colmoneill commented:
I hope I didn't jinx myself?

jancborchardt commented:
@colmoneill can you squash the changes into 2 commits? One making the actual changes, and one for only the whitespace/indentation change? That will make reviewing a lot easier.

irgendwie commented:
Took a quick look @colmoneill, looks good!

colmoneill commented:
Ok, I rebased.

Xenopathic commented:
I have a feeling this is broken... 

![screenshot - 050515 - 12 41 58](https://cloud.githubusercontent.com/assets/2016878/7471870/35de8428-f324-11e4-8cf3-62cf49d786d5.png)

jancborchardt commented:
Moving to 0.2 as it is a small fix with big potential for breakage. ;)

colmoneill commented:
Waw, I'm making a really big deal out of this...

@Xenopathic what browser are you on? I checked in Chromium, Firefox, Midori and Arora with all of those looking as planned.

Xenopathic commented:
@colmoneill That was Firefox on Linux, can't remember what version

colmoneill commented:
Shoot, I seem to have left my changes on the master branch of my pull request. I get what you see @Xenopathic when on the 'center-settings-form' branch, could that be the issue?

Sorry not used to submitting pull requests...

What should I do to make this integrate better?

Xenopathic commented:
@colmoneill You submitted this pull request from your master branch, so check the commits there and see if they are right. You might need to `git push`?

colmoneill commented:
No everything is up to date on 'master'. I'm just confused as to why you get bug displayed in the latest screenshot you posted.

On branch center-settings-form:
![screenshot from 2015-05-13 22 34 04](https://cloud.githubusercontent.com/assets/2122318/7620476/68b75efe-f9c0-11e4-9d69-c66c423376c1.png)
just like you posted,

but on branch master:
![screenshot from 2015-05-13 22 34 20](https://cloud.githubusercontent.com/assets/2122318/7620466/579d7d7e-f9c0-11e4-98e9-0d57e9ad489e.png)
all seems well

Again, apologies for my newness to this concept. I guess it's not impossible that something else is causing this, but as I can reproduce exactly the error just by switching branch, it's lightly to be this, right?

Xenopathic commented:
Hmm... I retested it and it works fine. Looking back at my reflog, I can see that I tested your `center-settings-form` branch, so that explains it.

Protip: please do work like this on a branch *not* called master :smile:

Anyway, :+1: from me after this gets a rebase and squash down to one commit.

colmoneill commented:
I hope I did that properly?

Xenopathic commented:
@colmoneill Unfortunately, no :(

After you've reverted those last commits (`git reset --hard e3288033f52ba5f0f22a1950b357a52d32ed680c`) you need to run rebase your branch onto ownCloud's master. This can be done by adding this repo as a new remote, then fetching that remote and finally rebasing it:

    git remote add owncloud https://github.com/owncloud/mail
    git fetch owncloud
    git rebase owncloud/master

You *will* get rebase conflicts - you'll need to fix them and follow the instructions git gives you. Eventually you will end up with your commits on top of the real master branch. Make sure they all look good!

Now, squash the commits together into one:

    git rebase -i HEAD^^^^^^
    (change all commits but the first one from 'pick' to 'squash')

You'll now be prompted to edit the commit message of your squashed commit. Finally, just force push back to your branch (force is needed since we rewrote history with the rebase).

colmoneill commented:
Thank you for your patience @Xenopathic it's much appreciated.

I followed your instructions, dealt with the conflict in the markup, and seems like all is good.

Cheers.

Xenopathic commented:
Assuming the changes still look correct (see https://github.com/owncloud/mail/pull/580/files?w=1 for a diff that hides the whitespace changes) then it's all gone well!

colmoneill commented:
If that was meant for me to respond then yes, that's the essence of my changes for this tiny thing :) Don't know if I can do anything else to help this get reviewed?

Xenopathic commented:
Well, my :+1: still holds. @jancborchardt What do you think? Refer to the no-whitespace diff for the real changes (which are very small)

jancborchardt commented:
@biolumin @irgendwie @clementhk can you review this? :)

irgendwie commented:
LGTM! :+1:

jancborchardt commented:
Really nice! Looks much better than currently.

Only one detail – there is a bit less space to the top than at the moment. Can you fix that? :) Then it’s golden and will be merged.

colmoneill commented:
Ok so I added a fixed padding to center the form vertically.
Git wise, I followed pretty much the same instructions that I was given previously, and it seems to be all good?
I fetched from mail origin, I rebased. I committed and I merged with the latest I could find. I didn't squash all commits as they were not mine. Not sure what is best to do in that case?
Hope this is ok to merge :)

biolumin commented:
@jancborchardt I'm only familiar with the front end: Javascript, HTML and CSS. I have yet to study the PHP and l10n stuff. At first glance it looks good, but reviewing this is out of my scope.

Was changing the index.html and mail.css effecting other back end files? I edited these two from the front end, and it resolved the issue of centering the form.

jancborchardt commented:
@Xenopathic your git help is needed again. ;)

jancborchardt commented:
@biolumin yeah -  if the commits are correctly squashed and it’s only @colmoneill’s additions, it’s only HTML and CSS.:)

Xenopathic commented:
Eeek, looks like you did a `git merge` @colmoneill :fearful: Thankfully, easy enough for fix:

```
git reset --hard 0888649e64052d8548b7871d1e43de74f7ec0089
git rebase origin/master # change this branch to the owncloud/mail master branch
# fix conflicts
git rebase --continue
git log # check history, make sure it looks OK with only your commit at the top
git push --force
```

colmoneill commented:
Ok thanks again @Xenopathic 

Is it ok to leave it like this for review? Should I squash that last commit also?

Cheers

jancborchardt commented:
@colmoneill yes, squash one more time please ;)

colmoneill commented:
Voilà? I hope?

Xenopathic commented:
That's another merge commit... :cry:

```
git reset --hard 090fed7ebd0616735b5d0a21db59f115b5a87d0b
git rebase -i HEAD^^
# set the second commit to 'squash' instead of 'pick'
# fix up commit message
git push --force
```

PLEASE never run `git merge` unless you know what you are doing with it :laughing:

colmoneill commented:
This is shameful... Well I'm learning lots, personally. I'm just costing you guys a lot of time...

Xenopathic commented:
:+1: That looks good to me!

colmoneill commented:
For sure! I do get it, never really had to delve this deep into Git, so I'm very grateful.
I'll be good to go for my next pull request for oC Mail!

irgendwie commented:
Found a small issue: the "connect-loading" gif is dislocated when you create a new Account (settings pane on the left open).

colmoneill commented:
Thanks @irgendwie I hadn't seen that!

For the CSS focused people reading: I kept the position of this one to absolute, to be able to land the spinner right in the middle of the 3 initial fields. The height is set absolutely, but the lateral positioning is set with this centering 'hack' of:
```
left: 0px;
right: 0px;
margin-left: auto;
margin-right auto;
```
which I find ok, but might not be up to standard for you? (or for Codacy?) If there is an other method for combining relative lateral positioning and absolute vertical positioning, I'll gladly take this on :)

irgendwie commented:
That was quick @colmoneill! :-)
LGTM now :+1:

colmoneill commented:
I had a small Codacy correction to make here, but I have no idea how I can help Scrutinizer along. @jancborchardt do you know who sets these checks up? Scrutinizer says:
```
The code coverage data was not received within the specified time. Please make sure your third-party service is configured to send code coverage data.
```
Any ideas?

Xenopathic commented:
@colmoneill Oh, it does that from time to time. Nothing to worry about, assuming the rest of the Scrutinizer report is OK. The important one is Travis CI unit tests - those passed, so the changes are good.

jancborchardt commented:
Nice @colmoneill :) welcome to ownCloud Mail, also hang around in our channel #owncloud-mail ;)

colmoneill commented:
Thanks for all the tips. See you out there.
